### ISSUE TITLE
Adapter for WordPress  wp-config

### ISSUE THREAD

**Issue by planetahuevo** [sebastianfeldmann/phpbu#271]
Hi, 
I am trying to use the adapter functionality to read the database user and password from the wp-config file on a WordPress installarion.
I have tested with the three types: env dotenv and array 
env do not work, which is normal. 
array do not return any errores but the whole thing does not work. 
dotenv should be the correct one, I think, but it returns an error here
"Failed to parse dotenv file due to an invalid name. Failed at [define( SECURE_AUTH_KEY, [edited]." 

Would be possible to create another adapter for WordPress that read the wp-config variables: 

```
define( 'DB_NAME', 'dbname' );
/** MySQL database username */
define( 'DB_USER', 'username' );
/** MySQL database password */
define( 'DB_PASSWORD', 'securepassword' );
```

Thanks!

**Comment by planetahuevo** [comment]
I think it should be as easy as creating a new Adapter like the phpArray but with this code on get value, but not sure how to make this into the phar:  
```
/**
     * Return a value for a given path.
     *
     * @param  string $path
     * @return string
     * @throws \phpbu\App\Exception
     */
    public function getValue(string $path) : string
    {
        return (string) constant($path);
    }
```

**Comment by sebastianfeldmann** [comment]
You would have to include the WordPress config but if  that's how WordPress handles its config then that should be it :)

You can open a pull request if you want ;)

**Comment by planetahuevo** [comment]
Never done a pull request before... I am not a developer myself but I suppose I can start with this one :) 
I was thinking to test it before to do any kind of pull request, but not sure how to put into the phar.
I will try to find out how to test it and then I will try to send the pull request. Please review it. I do not want o break something. :D

**Comment by planetahuevo** [comment]
For reference, this is the WordPress config file: 
https://github.com/WordPress/WordPress/blob/master/wp-config-sample.php

**Comment by sebastianfeldmann** [comment]
You do not have to put it into the PHAR, you can register Plugins in a `--bootstrap` file :)

I can do the Adapter, but I will most likely not get to it before end of next or even beginning of the week after that.
So if you can wait ... ;)

**Comment by sebastianfeldmann** [comment]
Thanks, I just published a new release 6.0.11 containing the `constants` adapter.

```xml
  <adapters>
    <adapter type="constants" name="wordpress">
      <option name="file" value="wordpress-config.php"/>
    </adapter>
  </adapters>
```

Hope that works ;)

**Comment by planetahuevo** [comment]
You are the best! 
I will test that!
Thank you very much.

**Comment by planetahuevo** [comment]
Oh, I am testing it on more sites and I found an issue. 
At the bottom of the file there is this line
`require_once ABSPATH . 'wp-settings.php';`

Normally wp-config.php and wp-settings.php will be on the same folder. 
but WordPress allow to put wp-config.php on a folder above the root, for security reasons. 
When that is happening, phpbu cannot find "wp-settings.php" because it look for it on the same folder. 
How  can I prevent this from happening. 
Is it possible to ignore that line when loading the file?

**Comment by planetahuevo** [comment]
I think we need another adapter specific for WordPress. This one is ok with standard constants files that do not have other includes or "require_once" like WordPress. 
Now that I know how to build them, I will test it and let you know.
Thanks

**Comment by planetahuevo** [comment]
So I created a new specific adapter for WordPress and added it as a Pull Request. :) 
Not sure the pull request is ok, as I needed to merge your changes before and I am way out of my comfort zone today with github. 
I have tested it on my install and it works fantastic (and I learned how to bootstrap files and extend the phar ;)). 
For the docs (not sure where to add this): 
The path needs to be one of these 4: 
DB_NAME, DB_USER, DB_PASSWORD or DB_HOST, or it will fail. 
As this is mainly for using mysqldump, it is more than enough, it could be extended in the future if someone needs other constants.

**Comment by sebastianfeldmann** [comment]
Great I will release an update in a sec

**Comment by sebastianfeldmann** [comment]
Wordpress adapter is released with version 6.0.12 :)

**Comment by planetahuevo** [comment]
Thanks!
